src/aabb.rs

```diff
@@ -141,6 +141,10 @@ pub trait Aabb: Sized {
         Aabb::new(self.min() + v, self.max() + v)
     }
 
+    /// Add a margin of the given width around the AABB, returning a new AABB.
+    #[inline]
+    fn add_margin(&self, margin: Self::Scalar) -> Self;
```

Comment:
we could take `Self::Vector` and implement it in the trait?

Reply:
That's possible yes. I don't mind either way.

Reply:
I'd prefer it to be a vector, assuming there are ways to turn scalar into a vector easily, if needed for the old API to be convenient

Reply:
There is, yes. Array trait in cgmath.

Reply:
I'll update it tonight